dvc/commands/experiments/show.py

```diff
@@ -65,6 +65,9 @@ def _collect_names(all_experiments, **kwargs):
 def _collect_rows(
     base_rev,
     experiments,
+    all_headers,
+    metric_headers,
+    param_headers,
```

Comment:
Would it be possible to just use `TabularData.row_from_dict()` here? The parameters are getting complicated here.

Reply:
Added change using `row_from_dict` but it only saves us from passing `all_headers`. 

Unfortunately, we still need `metric_headers` and `param_headers` later in this function because of the bug being fixed. 

The overall handling of metrics and params could be simplified but feels out of the scope of this fix.

Reply:
Makes sense. Thanks.

Reply:
We should simplify these parameters soon though, as it is getting complicated as `exp show` is becoming complex.
May get more confusing in the future. Anyway, thanks for looking into it. 🙂